packages/graph/hash_graph/lib/graph/src/shared/subgraph/edges/mod.rs

```diff
@@ -1,175 +1,130 @@
+    knowledge_graph: HashMap<EntityEditionId, HashSet<KnowledgeGraphOutwardEdges>>,
 }
 
-#[derive(Default, Debug, Serialize)]
-#[serde(transparent)]
-pub struct KnowledgeGraphRootedEdges(
-    // TODO: expose it through methods instead of making this field `pub`
-    //   see https://app.asana.com/0/1202805690238892/1203358665695491/f
-    pub HashMap<EntityId, HashMap<EntityVersion, HashSet<KnowledgeGraphOutwardEdges>>>,
-);
-
-// This is needed because Utoipa doesn't know how to handle the BTreeMaps
-impl ToSchema for KnowledgeGraphRootedEdges {
-    fn schema() -> openapi::Schema {
-        openapi::ObjectBuilder::new()
-            .additional_properties(Some(openapi::Schema::from(
-                openapi::ObjectBuilder::new().additional_properties(Some(openapi::Array::new(
-                    openapi::Ref::from_schema_name("KnowledgeGraphOutwardEdges"),
-                ))),
-            )))
-            .into()
+impl Serialize for Edges {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        crate::api::utoipa::subgraph::Edges {
+            ontology: crate::api::utoipa::subgraph::OntologyRootedEdges(self.ontology.iter().fold(
+                HashMap::new(),
+                |mut map, (id, edges)| {
+                    let edges = edges.iter().collect();
+                    match map.entry(id.base_id()) {
+                        Entry::Occupied(mut entry) => {
+                            entry.get_mut().insert(id.version(), edges);
+                        }
+                        Entry::Vacant(entry) => {
+                            entry.insert(HashMap::from([(id.version(), edges)]));
+                        }
+                    }
+                    map
+                },
+            )),
+            knowledge_graph: crate::api::utoipa::subgraph::KnowledgeGraphRootedEdges(
+                self.knowledge_graph
+                    .iter()
+                    .fold(HashMap::new(), |mut map, (id, edges)| {
+                        let edges = edges.iter().collect();
+                        match map.entry(id.base_id()) {
+                            Entry::Occupied(mut entry) => {
+                                entry.get_mut().insert(id.version(), edges);
+                            }
+                            Entry::Vacant(entry) => {
+                                entry.insert(HashMap::from([(id.version(), edges)]));
+                            }
+                        }
+                        map
+                    }),
+            ),
+        }
+        .serialize(serializer)
     }
 }
 
-#[derive(Default, Debug, Serialize)]
-pub struct Edges {
-    #[serde(flatten)]
-    ontology: OntologyRootedEdges,
-    #[serde(flatten)]
-    knowledge_graph: KnowledgeGraphRootedEdges,
+pub enum Edge {
+    Ontology {
+        edition_id: OntologyTypeEditionId,
+        edge: OntologyOutwardEdges,
```

Comment:
Can we call this `outward_edge` please

Reply:
Fixed in aeecf4a9275ce0d97cdf3f7071711aaa13b4ed4f